internals/overlord/logstate/gatherer.go

```diff
@@ -70,6 +70,10 @@ type logGatherer struct {
 	// ensure the client is not blocking subsequent teardown steps.
 	clientCancel context.CancelFunc
 
+	// Channels used to notify the main loop to update the client
+	flushCh   chan struct{}
+	setLabels chan svcWithLabels
```

Comment:
I'm inclined to disagree with you here - to me this feels like throwing away type safety just to save a few lines of code. But it's irrelevant anyway since we're removing the flushCh.

Reply:
I'm inclined to disagree with you here - to me this feels like throwing away type safety just to save a few lines of code. But it's irrelevant anyway since we're removing the flushCh.

Reply:
I don't understand how it's throwing away type safety when the recommendation is to literally use a type for each command?

Reply:
We can switch to a command interface + channel if there are more command types, but for now this is moot as we've gotten rid of the `flushCh` (and are doing the flushes "in line").